src/controllers/wallet/wallet.controller.js

```diff
@@ -355,6 +355,12 @@ async function decodeTx(req, res) {
 
     for (const output of tx.outputs) {
       output.parseScript(req.wallet.getNetworkObject());
+
+      let mine = false;
+      if (output.decodedScript) {
+        mine = await req.wallet.isAddressMine(output.decodedScript.address.base58);
```

Comment:
Will this work for data outputs? We have the following code for data outputs:

```
 outputData.decoded = {
  data: output.decodedScript.data,
};
```

So I believe that we will run this code for data outputs, and `output.decodedScript.address` will be `undefined`, won't it? In this case, `output.decodedScript.address.base58` would throw an error